rust-lib/src/lib.rs

```diff
@@ -358,21 +359,26 @@ pub extern "C" fn transfer(
 
 fn _accept(
     runtime: &COpaqueStruct,
-    consignment_bytes: *mut c_char,
+    consignment_bytes: *const u8,
+    consignment_length: c_int,
     reveal_outpoints: *mut c_char,
 ) -> Result<(), RequestError> {
     let runtime = Runtime::from_opaque(runtime)?;
 
-    let c_consignment_bytes = unsafe { CStr::from_ptr(consignment_bytes) };
-    let consignment_bytes: &[u8] = c_consignment_bytes.to_bytes();
+    let consignment_bytes = unsafe {
+        assert!(!consignment_bytes.is_null());
```

Comment:
Can we return a error here instead of panicking the whole program?

Reply:
I added an `Input` error to the `RequestError`. if you have a better name to propose, you're welcome